main.nf

```diff
@@ -1190,8 +1255,80 @@ process bam_trim {
     """
 }
 
+ch_pop_gen = Channel.create()
+ch_bam_for_pop_gen = Channel.empty()
+
+if(!params.skip_deduplication){
+    ch_bam_for_pop_gen.mix(ch_for_pop_gen,ch_trimmed_bam_for_genotyping).set {ch_pop_gen}
+} else {
+    ch_bam_for_pop_gen.mix(ch_bam_filtered_popgen,ch_for_pop_gen,ch_trimmed_bam_for_genotyping).set {ch_pop_gen}
+}
+
+if(!params.pop_gen){
+    ch_pop_gen.close()
+}
+
+
+process sex_determination {
+
+    when: params.pop_gen
+
+    input:
+    file(bam_file) from ch_pop_gen
```

Comment:
if it just takes a bamfile, you could simply use any bamfile from any upstream tool and mix the channels.